### PR TITLE
Explicitly bind req/res to context domain in express requestHandler

### PR DESCRIPTION
**Pull request by LewisJEllis**
Managed to repro/test that this fixes issue in #266. The test case is a little tricky but hopefully makes sense.

/cc @benvinegar @MaxBittker @rwky

### CODE REVIEW

**Comment by rwky** [comment]
I noticed you put a fix in for node 0.10, this and 0.12 aren't supported by the node core team https://github.com/nodejs/LTS no harm in adding in this case since it's just a test case but in general might not be worth the effort of supporting node < v4

**Comment by LewisJEllis** [comment]
Yea, we're certainly not holding back any new development/capabilities for the sake of node < v4, but we know we have a number of users still on those older versions (despite node's guidelines/pleading), so we support them as long as long as it's not a blocker/overly obnoxious to do so (which thus far it hasn't been).

The plan is roughly that when async_hooks shows up in node core, we'll do a new major version that supports only versions of node having async_hooks, and keep the older major version maintenance-only for everything pre-async_hooks.

File: test/raven.client.js [hunk 1]
```diff
@@ -1018,4 +1018,31 @@ describe('raven.middleware', function () {
...
+    var client = new raven.Client(dsn).install();
+
+    var EventEmitter = require('events');
+    // node 0.10 compat
+    if (process.version <= 'v0.11') EventEmitter = EventEmitter.EventEmitter;
+    var e = new EventEmitter();
+    e.on('done', function () {
+      // Context won't propagate here without the explicit binding of req/res done in the middleware
+      setTimeout(function () {
+        client.getContext().breadcrumbs.length.should.equal(1);
```
Comment (MaxBittker) [inline test/raven.client.js]: I assume that breadcrumbs[0].mesage==='test breadcrumb' here? is there a chance that some other one got added and this doesn't work?

Comment (LewisJEllis) [inline test/raven.client.js]: Yea, good call, I added an explicit check for that.

**Pull request merged**